### PR TITLE
shared.unattended: Allow ssh root login on Ubuntu 14.04 Linux server

### PR DESCRIPTION
**Pull request by spcui**
By default the root ssh login to Ubuntu 14.04 Linux
server is disable, need allow it.

Signed-off-by: Shuping Cui <scui@redhat.com>

### CODE REVIEW

**Comment by spcui** [comment]
ID: 1222836

**Comment by xutian** [comment]
LGTM,

Acked-by: XuTian <xutian@redhat.com>

File: shared/unattended/Ubuntu-14-04.preseed [hunk 1]
```diff
@@ -50,4 +50,5 @@ echo "start on stopped rc or RUNLEVEL=[2345]" > /target/etc/init/ttyS0.conf; \
 echo "stop on runlevel [!2345]" >> /target/etc/init/ttyS0.conf; \
 echo "respawn exec /sbin/getty -L 115200 ttyS0 vt102" >> /target/etc/init/ttyS0.conf; \
 echo "GRUB_RECORDFAIL_TIMEOUT=0" >> /target/etc/default/grub; \
-sed -i "s/ alias/ #alias/g" /target/root/.bashrc; in-target update-grub
+sed -i "s/ alias/ #alias/g" /target/root/.bashrc; in-target update-grub; \
+sed -i "s/$PermitRootLogin without-password/$PermitRootLogin yes/g" /target/etc/ssh/sshd_config
```
Comment (suqinhuang) [inline shared/unattended/Ubuntu-14-04.preseed]: Hi scui,
Why do we need "$" here.

BR
Suqin

Comment (spcui) [inline shared/unattended/Ubuntu-14-04.preseed]: means the beginning of the line.

Comment (suqinhuang) [inline shared/unattended/Ubuntu-14-04.preseed]: Hi scui,
The output will be "#PermitRootLogin yes" if it begin with "#"

**Comment by spcui** [comment]
# ./tools/scan_results.py 
Test                                                                                                                                                                                                         Status     Seconds  Info
----                                                                                                                                                                                                         ------     -------  ----
        (Result file: /root/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.4096m.repeat1.pre_test.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.Ubuntu.14.04-server.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_native    GOOD       977      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.Ubuntu.14.04-server.x86_64.io-github-autotest-qemu.boot                                                                  GOOD       14       completed successfully
----                                                                                                                                                                                                         GOOD       996

**Comment by CongSmile** [comment]
Could you please paste some other cases which covered the reboot in the test?

Thanks.

File: shared/unattended/Ubuntu-14-04.preseed [hunk 2]
```diff
@@ -50,4 +50,5 @@ echo "start on stopped rc or RUNLEVEL=[2345]" > /target/etc/init/ttyS0.conf; \
 echo "stop on runlevel [!2345]" >> /target/etc/init/ttyS0.conf; \
 echo "respawn exec /sbin/getty -L 115200 ttyS0 vt102" >> /target/etc/init/ttyS0.conf; \
 echo "GRUB_RECORDFAIL_TIMEOUT=0" >> /target/etc/default/grub; \
-sed -i "s/ alias/ #alias/g" /target/root/.bashrc; in-target update-grub
+sed -i "s/ alias/ #alias/g" /target/root/.bashrc; in-target update-grub; \
+sed -i "s/^PermitRootLogin without-password/PermitRootLogin yes/g" /target/etc/ssh/sshd_config
```
Comment (suqinhuang) [inline shared/unattended/Ubuntu-14-04.preseed]: Acked-by: Suqin Huang

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by spcui** [comment]
Hi @CongSmile 

# ./tools/scan_results.py 
Test                                                                                                                                                  Status     Seconds  Info
----                                                                                                                                                  ------     -------  ----
        (Result file: /root/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.1.qcow2.virtio_blk.up.virtio_net.Ubuntu.14.04-server.x86_64.io-github-autotest-qemu.format_disk    GOOD       22       completed successfully
----                                                                                                                                                  GOOD       25

**Comment by xutian** [comment]
Merged, thanks @spcui  and @suqinhuang

**Pull request merged**